### DISCUSSION TITLE
How do groups really work?

### DISCUSSION THREAD

**Discussion by klapauciusisgreat** [Koenkk/zigbee2mqtt#20425] [General]
I would really like to understand how groups work under the hood.

I have two zigbee relais and a scene switch. I want to use one of the 4 switches on the scene switch to toggle BOTH relais.

I have created a group in the UI, then bound one of the scene switch genOnOff endpoints to the group.

When I toggle the scene switch, I can see a MQTT message for the scene switch, but the group does nothing, and also does not show up in the logs (maybe in the herdsman logs, which are too cryptic for me to properly understand)

Now, I lack a fundamental understanding of groups, it seems.

1) Is the group something you set in the relais itself, and does the scene switch binding to a group actually program the switch so it sends a group broadcast message ? In that case, I believe that this switch would work even if the z2m controller were to stop completely ?

Or,

2) Is the group message created by the controller (eg herdsman ?) Ie, the controller sees the sceneswitch toggle and it creates either a broadcast message ot unicast messages to toggle the switches ?

3) Is there a way in the UI to toggle a group directly ? I can look at the group and see the group member and can trigger them directly, but I found no UI element to trigger a group itself.

4) I noticed that binding to the default group 0 does not seem to work - the UI just greys out the bind button. Is that a bug or feature ?

5) Do certain switches just don't support group binding at the hardware/firmware level ?

I tried to read the manuals but didn't really find a succinct answer to these questions. If they are there, I would really love a pointer, if not, I'd still like a pointer (maybe to the relevant code - I admit I have been putting off looking at the code out of lazyness)

I can detail my setup, but I believe my questions are too general for that.

Many thanks in advance, and Happy New Year

**Comment by Koenkk** [comment]
> Is the group something you set in the relais itself, and does the scene switch binding to a group actually program the switch so it sends a group broadcast message ? In that case, I believe that this switch would work even if the z2m controller were to stop completely ?

Yes. Messages to groups are broadcast messages, they are received by every device in the network. When received by the device, it check wether it is part of that group, if it is, applies the command to itself.

> Is the group message created by the controller (eg herdsman ?) Ie, the controller sees the sceneswitch toggle and it creates either a broadcast message ot unicast messages to toggle the switches ?

It can be created by both, any device in the network or by the controller by sending a MQTT message to that group ([docs](https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html#zigbee2mqtt-friendly-name-set))

> Is there a way in the UI to toggle a group directly ? I can look at the group and see the group member and can trigger them directly, but I found no UI element to trigger a group itself.

Currently not supported by the UI, but can be published via MQTT, see previous answer 

> I noticed that binding to the default group 0 does not seem to work - the UI just greys out the bind button. Is that a bug or feature ?

Don't use this group, as some devices subscribe to it by default

> Do certain switches just don't support group binding at the hardware/firmware level ?

Yes, for my experience:
- Receiving side: work almost always (e.g. switches, plugs)
- Sending side: hit and miss, either group binding is not supported or it doesn't send standard commands (e.g. remotes)

**Comment by thogens** [comment]
I've been using Zigbee groups for some time and also was using Zigbee scenes, but could it be that this constant broadcasts can slow down the network? I'm currently migrating away from Zigbee groups to avoid flooding the network, but maybe it isn't really necessary...
wdyt?

**Reply by thogens** [reply]
I'd like to reply to myself here. Moving away from Zigbee groups was actually not the best idea, as this results in 30x commands if you manipulate a "HA group", which can overload the Zigbee stack. So moving back to zigbee groups in most cases.
